### ISSUE TITLE
RingbufferAddReadOneStressTest.whenLongTTLAndSmallBuffer

### ISSUE THREAD

**Issue by Donnerbart** [hazelcast/hazelcast#7339]
Failed on a local run for code coverage.
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-1 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:129153 is too small. The current headSequence is:129292 tailSequence is:130291>

	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:62)
	at com.hazelcast.ringbuffer.impl.RingbufferAddReadOneStressTest.test(RingbufferAddReadOneStressTest.java:82)
	at com.hazelcast.ringbuffer.impl.RingbufferAddReadOneStressTest.whenLongTTLAndSmallBuffer(RingbufferAddReadOneStressTest.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
```
https://gist.githubusercontent.com/Donnerbart/a3f47e3bdd0570b5eacc/raw/1b98c60aac9b857968ee2f15c301251c4c345c70/ringbuffer_failure.log

**Comment by Donnerbart** [comment]
```
```
18:13:44,044  INFO |whenLongTTLAndSmallBuffer| - [OperationService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Backpressure is disabled
18:13:44,072  INFO |whenLongTTLAndSmallBuffer| - [ClassicOperationExecutor] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Starting with 4 generic operation threads and 8 partition operation threads.
18:13:44,150 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Updating members [Member [127.0.0.1]:10217 this]
18:13:44,150  INFO |whenLongTTLAndSmallBuffer| - [system] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Hazelcast 3.6-SNAPSHOT (20160104) starting at Address[127.0.0.1]:10217
18:13:44,150  INFO |whenLongTTLAndSmallBuffer| - [system] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
18:13:44,150  INFO |whenLongTTLAndSmallBuffer| - [system] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Configured Hazelcast Serialization version : 1
18:13:44,151  INFO |whenLongTTLAndSmallBuffer| - [LifecycleService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Address[127.0.0.1]:10217 is STARTING
18:13:44,300  INFO |whenLongTTLAndSmallBuffer| - [MockJoiner] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] 


Members [1] {
	Member [127.0.0.1]:10217 this
}

18:13:44,300  WARN |whenLongTTLAndSmallBuffer| - [Node] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Config seed port is 5701 and cluster size is 1. Some of the ports seem occupied!
18:13:44,317  INFO |whenLongTTLAndSmallBuffer| - [LifecycleService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Address[127.0.0.1]:10217 is STARTED
18:13:44,360  INFO |whenLongTTLAndSmallBuffer| - [OperationService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Backpressure is disabled
18:13:44,393  INFO |whenLongTTLAndSmallBuffer| - [ClassicOperationExecutor] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Starting with 4 generic operation threads and 8 partition operation threads.
18:13:44,492 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Updating members [Member [127.0.0.1]:10218 this]
18:13:44,493  INFO |whenLongTTLAndSmallBuffer| - [system] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Hazelcast 3.6-SNAPSHOT (20160104) starting at Address[127.0.0.1]:10218
18:13:44,493  INFO |whenLongTTLAndSmallBuffer| - [system] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
18:13:44,493  INFO |whenLongTTLAndSmallBuffer| - [system] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Configured Hazelcast Serialization version : 1
18:13:44,493  INFO |whenLongTTLAndSmallBuffer| - [LifecycleService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Address[127.0.0.1]:10218 is STARTING
18:13:44,683  INFO |whenLongTTLAndSmallBuffer| - [MockConnectionManager] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Created connection to endpoint: Address[127.0.0.1]:10217, connection: MockConnection{localEndpoint=Address[127.0.0.1]:10218, remoteEndpoint=Address[127.0.0.1]:10217}
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterJoinManager] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Handling join from Address[127.0.0.1]:10218, joinInProgress: false
18:13:44,684  INFO |whenLongTTLAndSmallBuffer| - [MockConnectionManager] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Created connection to endpoint: Address[127.0.0.1]:10218, connection: MockConnection{localEndpoint=Address[127.0.0.1]:10217, remoteEndpoint=Address[127.0.0.1]:10218}
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterJoinManager] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Starting join...
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterJoinManager] hz._hzInstance_5550_dev.generic-operation.thread-0 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Handling master response Address[127.0.0.1]:10217 from Address[127.0.0.1]:10217
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterHeartbeatManager] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] MasterConfirmation has been received from Member [127.0.0.1]:10218
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Updating members [Member [127.0.0.1]:10217 this, Member [127.0.0.1]:10218]
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterJoinManager] hz._hzInstance_5549_dev.generic-operation.thread-1 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Join is in progress, cannot handle join request from Address[127.0.0.1]:10218 at the moment
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-1 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-1 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterHeartbeatManager] hz._hzInstance_5550_dev.generic-operation.thread-1 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] MasterConfirmation has been received from Member [127.0.0.1]:10217
18:13:44,684 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-1 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Updating members [Member [127.0.0.1]:10217, Member [127.0.0.1]:10218 this]
18:13:44,711  INFO |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-1 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] 

Members [2] {
	Member [127.0.0.1]:10217
	Member [127.0.0.1]:10218 this
}

18:13:44,711  INFO |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] 

Members [2] {
	Member [127.0.0.1]:10217 this
	Member [127.0.0.1]:10218
}

18:13:44,711 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-2 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to -12ms.
18:13:45,735  WARN |whenLongTTLAndSmallBuffer| - [Node] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Config seed port is 5701 and cluster size is 2. Some of the ports seem occupied!
18:13:45,748  INFO |whenLongTTLAndSmallBuffer| - [LifecycleService] whenLongTTLAndSmallBuffer - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Address[127.0.0.1]:10218 is STARTED
Starting test
18:13:45,764  INFO |whenLongTTLAndSmallBuffer| - [RingbufferAddReadOneStressTest$ConsumeThread] ConsumeThread-1 - ConsumeThread-1 Starting
18:13:45,764  INFO |whenLongTTLAndSmallBuffer| - [InternalPartitionService] ConsumeThread-1 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] Initializing cluster partition table arrangement...
18:13:45,789  INFO |whenLongTTLAndSmallBuffer| - [RingbufferAddReadOneStressTest$ConsumeThread] ConsumeThread-2 - ConsumeThread-2 Starting
18:13:47,792  INFO |whenLongTTLAndSmallBuffer| - [RingbufferAddReadOneStressTest$ProduceThread] ProduceThread - ProduceThread Starting
ConsumeThread-1 at 355
ConsumeThread-2 at 1761
18:13:49,160 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-0 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ProduceThread at 6793
ConsumeThread-1 at 7031
ConsumeThread-2 at 8320
ProduceThread at 13024
ConsumeThread-1 at 13300
ConsumeThread-2 at 14719
ProduceThread at 19682
ConsumeThread-1 at 20033
18:13:54,161 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-2 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ConsumeThread-2 at 21455
ProduceThread at 25526
ConsumeThread-1 at 25742
ConsumeThread-2 at 27379
ProduceThread at 31024
ConsumeThread-1 at 31296
ConsumeThread-2 at 32397
18:13:59,162 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-3 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ProduceThread at 37474
ConsumeThread-1 at 37886
ConsumeThread-2 at 39674
ProduceThread at 45457
ConsumeThread-1 at 45621
ConsumeThread-2 at 47029
ProduceThread at 51756
ConsumeThread-1 at 51931
18:14:04,162 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-0 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ConsumeThread-2 at 53606
ProduceThread at 59298
ConsumeThread-1 at 59477
ConsumeThread-2 at 61175
ProduceThread at 66953
ConsumeThread-1 at 67134
ConsumeThread-2 at 68902
18:14:09,163 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-2 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ProduceThread at 74424
ConsumeThread-1 at 74556
ConsumeThread-2 at 76157
ProduceThread at 81539
ConsumeThread-1 at 81641
ConsumeThread-2 at 83324
ProduceThread at 88759
ConsumeThread-1 at 88940
18:14:14,163 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-3 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ConsumeThread-2 at 90442
18:14:14,508 TRACE |whenLongTTLAndSmallBuffer| - [ClusterHeartbeatManager] cached3 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Sending MasterConfirmation to Member [127.0.0.1]:10217
18:14:14,508 TRACE |whenLongTTLAndSmallBuffer| - [ClusterHeartbeatManager] hz._hzInstance_5549_dev.generic-operation.thread-0 - [127.0.0.1]:10217 [dev] [3.6-SNAPSHOT] MasterConfirmation has been received from Member [127.0.0.1]:10218
ProduceThread at 95490
ConsumeThread-1 at 95576
ConsumeThread-2 at 96810
ProduceThread at 102432
ConsumeThread-1 at 102623
ConsumeThread-2 at 103766
18:14:19,164 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-0 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ProduceThread at 108586
ConsumeThread-1 at 108757
ConsumeThread-2 at 110364
ProduceThread at 115522
ConsumeThread-1 at 115683
ConsumeThread-2 at 117324
ProduceThread at 122312
ConsumeThread-1 at 122491
18:14:24,164 TRACE |whenLongTTLAndSmallBuffer| - [ClusterService] hz._hzInstance_5550_dev.generic-operation.thread-2 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] Setting cluster time diff to 0ms.
ConsumeThread-2 at 124159
ProduceThread at 129678
ConsumeThread-1 at 129153
18:14:25,984 ERROR |whenLongTTLAndSmallBuffer| - [ReadOneOperation] hz._hzInstance_5550_dev.partition-operation.thread-7 - [127.0.0.1]:10218 [dev] [3.6-SNAPSHOT] sequence:129153 is too small. The current headSequence is:129292 tailSequence is:130291
com.hazelcast.ringbuffer.StaleSequenceException: sequence:129153 is too small. The current headSequence is:129292 tailSequence is:130291
	at com.hazelcast.ringbuffer.impl.RingbufferContainer.checkBlockableReadSequence(RingbufferContainer.java:177)
	at com.hazelcast.ringbuffer.impl.operations.ReadOneOperation.beforeRun(ReadOneOperation.java:46)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:166)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:393)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.processPacket(OperationThread.java:184)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.process(OperationThread.java:137)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.doRun(OperationThread.java:124)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.run(OperationThread.java:99)
18:14:25,986  WARN |whenLongTTLAndSmallBuffer| - [RingbufferAddReadOneStressTest$ConsumeThread] ConsumeThread-1 - ConsumeThread-1 Completed with failure
com.hazelcast.ringbuffer.StaleSequenceException: sequence:129153 is too small. The current headSequence is:129292 tailSequence is:130291
	at com.hazelcast.ringbuffer.impl.RingbufferContainer.checkBlockableReadSequence(RingbufferContainer.java:177)
	at com.hazelcast.ringbuffer.impl.operations.ReadOneOperation.beforeRun(ReadOneOperation.java:46)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:166)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:393)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.processPacket(OperationThread.java:184)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.process(OperationThread.java:137)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.doRun(OperationThread.java:124)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.run(OperationThread.java:99)
	at ------ End remote and begin local stack-trace ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveApplicationResponse(InvocationFuture.java:385)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveApplicationResponseOrThrowException(InvocationFuture.java:335)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.get(InvocationFuture.java:223)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.get(InvocationFuture.java:202)
	at com.hazelcast.ringbuffer.impl.RingbufferProxy.readOne(RingbufferProxy.java:154)
	at com.hazelcast.ringbuffer.impl.RingbufferAddReadOneStressTest$ConsumeThread.doRun(RingbufferAddReadOneStressTest.java:144)
	at com.hazelcast.test.TestThread.run(TestThread.java:27)
18:14:25,987  INFO |whenLongTTLAndSmallBuffer| - [RingbufferAddReadOneStressTest$ProduceThread] ProduceThread - ProduceThread Completed
18:14:25,987  INFO |whenLongTTLAndSmallBuffer| - [RingbufferAddReadOneStressTest$ConsumeThread] ConsumeThread-2 - ConsumeThread-2 Completed
Waiting for completion
```

**Comment by Donnerbart** [comment]
Failed again locally:
```
java.lang.AssertionError: No error should have been thrown, but ConsumeThread-2 completed error expected null, but was:<com.hazelcast.ringbuffer.StaleSequenceException: sequence:640700 is too small. The current headSequence is:640718 tailSequence is:641717>

	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at com.hazelcast.test.TestThread.assertSucceedsEventually(TestThread.java:62)
	at com.hazelcast.ringbuffer.impl.RingbufferAddReadOneStressTest.test(RingbufferAddReadOneStressTest.java:83)
	at com.hazelcast.ringbuffer.impl.RingbufferAddReadOneStressTest.whenLongTTLAndSmallBuffer(RingbufferAddReadOneStressTest.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by pveentjer** [comment]
Closing this issue since solved by #8384
